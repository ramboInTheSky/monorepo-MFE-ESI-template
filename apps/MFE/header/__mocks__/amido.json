{
  "id": "33d0153c-ed8b-4a65-b63c-5ad5edddd187",
  "name": "Default",
  "realm": "amido",
  "territory": "GB",
  "language": "en",
  "regions": [
    {
      "type": "Brand",
      "title": "Brand",
      "accessibilityTitle": "Brand",
      "elements": [
        {
          "elements": [],
          "type": "link",
          "url": "/",
          "openInNewWindow": false,
          "icon": "/icons/header/brand/amido/logo.svg",
          "text": "Amido Logo",
          "accessibilityText": "Amido Icon",
          "tooltip": "Amido Icon",
          "wideModeIcon": "/icons/header/brand/amido/logo.svg",
          "narrowModeIcon": "/icons/header/brand/amido/logo.svg"
        }
      ]
    },
    {
      "type": "QuickLinks",
      "title": "QuickLinks",
      "accessibilityTitle": "QuickLinks",
      "elements": [
        {
          "elements": [
            {
              "type": "MyAccountLoggedIn",
              "url": "/secure/accounts/transfer",
              "openInNewWindow": false,
              "icon": "/icons/header/quicklinks/amido/my-account.svg",
              "text": "<username>",
              "accessibilityText": "<username>",
              "tooltip": "<username>'s account",
              "wideModeIcon": "/icons/header/quicklinks/amido/my-account_desktop.svg",
              "narrowModeIcon": "/icons/header/quicklinks/amido/my-account.svg"
            },
            {
              "type": "MyAccountLoggedOut",
              "url": "/secure/accounts/transfer",
              "openInNewWindow": false,
              "icon": "/icons/header/quicklinks/amido/my-account.svg",
              "text": "My Account",
              "accessibilityText": "Sign in to view account details",
              "tooltip": "Sign in to view account details",
              "wideModeIcon": "/icons/header/quicklinks/amido/my-account_desktop.svg",
              "narrowModeIcon": "/icons/header/quicklinks/amido/my-account.svg"
            }
          ],
          "type": "MyAccount",
          "url": null,
          "openInNewWindow": true,
          "icon": null,
          "text": "MyAccount",
          "accessibilityText": "MyAccount",
          "tooltip": null,
          "wideModeIcon": null,
          "narrowModeIcon": null
        },
        {
          "elements": [],
          "type": "link",
          "url": "/help",
          "openInNewWindow": true,
          "icon": null,
          "text": "Help",
          "accessibilityText": "Help",
          "tooltip": "Help - FAQ's, how to shop online, sizing guides and more",
          "wideModeIcon": null,
          "narrowModeIcon": null
        },
        {
          "elements": [],
          "type": "link",
          "url": "/stores/",
          "openInNewWindow": false,
          "icon": null,
          "text": "Store Locator",
          "accessibilityText": "Store Locator",
          "tooltip": "Find your nearest Amido Store",
          "wideModeIcon": null,
          "narrowModeIcon": null
        }
      ]
    }
  ]
}
