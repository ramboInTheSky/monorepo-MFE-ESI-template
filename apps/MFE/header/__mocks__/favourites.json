{
  "data": {
    "ShoppingLists": [
      {
        "ListId": 39931807,
        "ListName": "Favourite List",
        "TypeId": 1,
        "DateAdded": "/Date(-62135596800000)/",
        "DateUpdated": "/Date(1621516301237)/",
        "DefaultList": true,
        "IsAuthenticated": false
      }
    ],
    "ShoppingListItems": [
      {
        "ItemNumber": "499-373",
        "OptionCode": "-1",
        "ItemSizeOptions": null,
        "Chain": null,
        "ItemUrl": null,
        "DateAdded": "/Date(-62135596800000)/",
        "DateUpdated": "/Date(-62135596800000)/",
        "ItemDescription": null,
        "SizeDescription": null,
        "LinkedItemNumber": "",
        "LinkedOptionNo": "",
        "ItemBrand": null,
        "ItemCategory": null,
        "ImageUrl": null,
        "FormattedPrice": null,
        "LightboxUrl": null,
        "Price": null,
        "MinPrice": null,
        "MaxPrice": null,
        "ListId": 39931807,
        "LegalWarning": null,
        "ItemSearchDescription": null,
        "Department": null,
        "Colour": null,
        "Fit": null,
        "StyleDescription": null,
        "PremiumSiteIndicator": null,
        "IsTradable": false
      }
    ],
    "MaximumLimit": 150,
    "ExceededLimitBy": 0,
    "InvalidItemsCount": 0,
    "ValidItemsCount": 0,
    "AuthenticatedUser": false,
    "IdentifiedUser": false,
    "FavouriteLoginPromptDisplayed": false,
    "HardLogoutPromptDisplayed": false,
    "SoftLoginFirstname": "",
    "LegalWarningBannerMessage": null
  },
  "Success": true
}
