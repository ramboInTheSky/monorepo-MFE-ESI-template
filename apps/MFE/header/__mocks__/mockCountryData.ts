export const mockCountriesData = [
    {
        Name: "nl",
        Country: "Netherlands",
        CountryCode: "NL",
        DisplayText: "Netherlands (€)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "nl",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.com/nl/en",
                AccountDomainUrl: "account.amido.com/nl/en",
            },
            {
                Name: "nl",
                Value: "Nederlands ",
                TargetUrl: "http://www.amido.com/nl/nl",
                AccountDomainUrl: "account.amido.com/nl/nl",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/nl.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 0,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "ae",
        Country: "United Arab Emirates",
        CountryCode: "AE",
        DisplayText: "United Arab Emirates (AED)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "en",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.ae/en",
                AccountDomainUrl: "account.amido.ae/en",
            },
            {
                Name: "ar",
                Value: "العربية ",
                TargetUrl: "http://www.amido.ae/ar",
                AccountDomainUrl: "account.amido.ae/ar",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/ae.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 0,
        Region: "Middle East / Africa",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "gb",
        Country: "United Kingdom",
        CountryCode: "GB",
        DisplayText: "United Kingdom (£)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "en",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "//www.amido.com",
                AccountDomainUrl: "www.amido.com/",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/gb.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 1,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "",
        Country: "British Forces Post Office",
        CountryCode: "",
        DisplayText: "British Forces Post Office ",
        NativeCountryText: "",
        RedirectUrl: "http://www.amido.com/bfpo",
        DefaultLanguageName: "",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.com/bfpo",
                AccountDomainUrl: "",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/bfpo.png",
        DisplaySequenceAttribute: "AAAAAAZ",
        PromotedFlagIndex: 0,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "test",
        Country: "Test Country",
        CountryCode: "TTT",
        DisplayText: "TEST COUNTRY",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "TTT",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.com/nl/en",
                AccountDomainUrl: "account.amido.com/nl/en",
            },
            {
                Name: "nl",
                Value: "Nederlands ",
                TargetUrl: "http://www.amido.com/nl/nl",
                AccountDomainUrl: "account.amido.com/nl/nl",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/nl.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 0,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: true,
        HideInPage: false,
        IsROW: false,
    },
]

export const mockCountriesDataWithoutBFPO = [
    {
        Name: "nl",
        Country: "Netherlands",
        CountryCode: "NL",
        DisplayText: "Netherlands (€)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "nl",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.com/nl/en",
                AccountDomainUrl: "account.amido.com/nl/en",
            },
            {
                Name: "nl",
                Value: "Nederlands ",
                TargetUrl: "http://www.amido.com/nl/nl",
                AccountDomainUrl: "account.amido.com/nl/nl",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/nl.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 0,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "ae",
        Country: "United Arab Emirates",
        CountryCode: "AE",
        DisplayText: "United Arab Emirates (AED)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "en",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.ae/en",
                AccountDomainUrl: "account.amido.ae/en",
            },
            {
                Name: "ar",
                Value: "العربية ",
                TargetUrl: "http://www.amido.ae/ar",
                AccountDomainUrl: "account.amido.ae/ar",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/ae.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 0,
        Region: "Middle East / Africa",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "gb",
        Country: "United Kingdom",
        CountryCode: "GB",
        DisplayText: "United Kingdom (£)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "en",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "//www.amido.com",
                AccountDomainUrl: "www.amido.com/",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/gb.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 1,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "test",
        Country: "Test Country",
        CountryCode: "TTT",
        DisplayText: "TEST COUNTRY",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "TTT",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.com/nl/en",
                AccountDomainUrl: "account.amido.com/nl/en",
            },
            {
                Name: "nl",
                Value: "Nederlands ",
                TargetUrl: "http://www.amido.com/nl/nl",
                AccountDomainUrl: "account.amido.com/nl/nl",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/nl.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 0,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: true,
        HideInPage: false,
        IsROW: false,
    },
]

export const mockCountriesDataWithBFPOCorrected = [
    {
        Name: "nl",
        Country: "Netherlands",
        CountryCode: "NL",
        DisplayText: "Netherlands (€)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "nl",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.com/nl/en",
                AccountDomainUrl: "account.amido.com/nl/en",
            },
            {
                Name: "nl",
                Value: "Nederlands ",
                TargetUrl: "http://www.amido.com/nl/nl",
                AccountDomainUrl: "account.amido.com/nl/nl",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/nl.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 0,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "ae",
        Country: "United Arab Emirates",
        CountryCode: "AE",
        DisplayText: "United Arab Emirates (AED)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "en",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.ae/en",
                AccountDomainUrl: "account.amido.ae/en",
            },
            {
                Name: "ar",
                Value: "العربية ",
                TargetUrl: "http://www.amido.ae/ar",
                AccountDomainUrl: "account.amido.ae/ar",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/ae.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 0,
        Region: "Middle East / Africa",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "gb",
        Country: "United Kingdom",
        CountryCode: "GB",
        DisplayText: "United Kingdom (£)",
        NativeCountryText: "",
        RedirectUrl: "",
        DefaultLanguageName: "en",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "//www.amido.com",
                AccountDomainUrl: "www.amido.com/",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/gb.png",
        DisplaySequenceAttribute: "",
        PromotedFlagIndex: 1,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
    {
        Name: "bfpo",
        Country: "British Forces Post Office",
        CountryCode: "BFPO",
        DisplayText: "British Forces Post Office ",
        NativeCountryText: "",
        RedirectUrl: "http://www.amido.com/bfpo",
        DefaultLanguageName: "en",
        Languages: [
            {
                Name: "en",
                Value: "English",
                TargetUrl: "http://www.amido.com/bfpo",
                AccountDomainUrl: "",
            },
        ],
        iconUrl: "/static-content/icons/shared/countryflags/bfpo.png",
        DisplaySequenceAttribute: "AAAAAAZ",
        PromotedFlagIndex: 0,
        Region: "Europe",
        DomainType: "",
        HideInDropdown: false,
        HideInPage: false,
        IsROW: false,
    },
]
