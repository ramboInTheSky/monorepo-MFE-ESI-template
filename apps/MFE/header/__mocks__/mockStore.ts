import configureMockStore from "redux-mock-store"
import thunk from "redux-thunk"
import {mockColors} from "@monorepo/themes"
import {State} from "../src/ducks"
import apiData from "./amido.json"
import {ShoppingBag, ChargesAndIncentive, Item} from "../src/models/shoppingbag"

export const mockStyles = {
    sm: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    xs: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    md: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
        margin: "10rem",
        decoration: "none",
    },
    lg: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    xl: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
}

export const mockMyAccountStyles = {
    ...mockStyles,
    SmallIcon: mockStyles,
    SmallIconContainer: mockStyles,
    IconOnly: mockStyles,
    IconContainer: mockStyles,
    ToolTip: mockStyles,
    Divider: mockStyles,
}

export const mockCountrySelectorStyles = {
    ...mockStyles,
    Container: mockStyles,
}

const mockShoppingBagStyles = {
    ...mockStyles,
    productImage: mockStyles,
    itemsContainer: mockStyles,
}

const mockFavouritesStyles = {
    ...mockStyles,
    img: mockStyles,
}

export const mockRealmStyles = {
    CTA: mockStyles,
    UpperHeader: mockStyles,
    Meganav: mockStyles,
    PrimaryNav: mockStyles,
    Brand: mockStyles,
    SearchBox: mockStyles,
    Checkout: mockStyles,
    Quicklinks: mockStyles,
    SearchBoxDivider: mockStyles,
    SearchBoxIcon: mockStyles,
    Favourites: mockFavouritesStyles,
    CountrySelector: mockCountrySelectorStyles,
    ShoppingBag: mockShoppingBagStyles,
    MyAccount: mockMyAccountStyles,
    AutoCompleteButton: mockStyles,
    DrawerContainer: mockStyles,
}

const discount = 10
export const mockShoppingBag = {
    itemCount: 2,
    loading: false,
    bag: {
        FirstName: "cypress-account",
        Description: "Excluding UK Standard Delivery (Normally £3.99)",
        MultiBuyDiscount: 1,
        FinalOrderValue: 89,
        FinalOrderValueFormatted: "£89.00",
        MultiBuyDiscountFormatted: "£1.00",
        ChargesAndIncentives: [
            {
                AdditionalAmount: -discount,
                AdditionalAmountFormatted: `-£${discount.toFixed(2)}`,
                AdditionalCode: "UKCO1",
                MaximumOrderValue: 0,
                MinimumOrderValue: 15,
                OfferShortDescription: "Welcome Offer",
                OfferTypeDescription: "Welcome Offer",
                Type: "PRO",
            } as ChargesAndIncentive,
        ],
        ItemCount: 2,
        Items: [
            {
                ItemID: 1,
                ItemNumber: "436441",
                OptionNo: "04",
                SizeDescription: "10 S",
                Price: "45.00",
                Quantity: 10,
                StockMessage: "Delayed",
                Description: "Black Lift, Slim And Shape Skinny Jeans",
                AlternativeDescription: "Alternative - Black Lift, Slim And Shape Skinny Jeans",
                CistStoreName: "",
                Url: "/path1",
            } as Item,
            {
                ItemID: 2,
                ItemNumber: "436642",
                OptionNo: "04",
                SizeDescription: "15 L",
                Price: "45.00",
                Quantity: 5,
                StockMessage: "In Stock",
                Description: "Black Lift, Slim And Shape Skinny Jeans 2",
                AlternativeDescription: "Alternative - Black Lift, Slim And Shape Skinny Jeans 2",
                CistStoreName: "LEICESTER - HIGH CROSS SHOPPING CENTRE",
                Url: "/path2",
            } as Item,
        ],
    } as ShoppingBag,
}

export const mockRecents = {
    queryIds: ["celio", "gucci", "prada", "&", "#", "+"],
}

export const mockAutoComplete: any = {
    isLoading: false,
    q: "socks",
    suggestions: [
        {
            dq: "girls socks",
            q: "girls socks",
        },
        {
            dq: "boys socks",
            q: "boys socks",
        },
    ],
    products: [],
    numFound: 2,
    parameters: {
        accountId: "abc",
        domainKey: "Amido",
    },
}

export const mockLanguages = {
    currentLanguageText: "English",
    altLanguageName: "ar",
    altLanguageUrl: "/ar",
    currentLanguageName: "en",
    direction: "ltr",
    siteUrl: "http://amido.com",
}

export const mockSearch = {
    typedCharacters: "",
    showAutocomplete: false,
    showRecentSearch: false,
}

export const mockText = {
    recentSearches: {
        enriched: {
            mobileHeader: "RECENT SEARCHES",
            desktopHeader: "Your recent searches: ",
        },
        simple: {
            mobileHeader: "RECENT SEARCHES",
            desktopHeader: "RECENT SEARCHES",
        },
    },
    searchBox: {
        label: "Search",
        bigPlaceholder: "Search product",
        smallPlaceholder: "Search...",
        clearButton: {
            alt: "Clear search button",
        },
        closeButton: {
            text: "Close",
        },
    },
    myAccount: {
        title: "View Account Summary",
        buttonText: "SIGN OUT",
        altText: "My account icon",
        tooltipAltText: "icon",
        notText: "Not",
    },
    checkout: {
        text: "Checkout",
    },
    autoComplete: {
        loadingText: "Loading...",
        noResults: "No results found for ",
        headerText: "TOP MATCHES",
        searchLinktext: "See all results",
        enrich: {
            termTitle: "Top results for ",
        },
    },
    drawer: {
        closeIconText: "Close icon",
    },
    favourites: {
        altText: "Favourites icon",
    },
    miniShoppingBag: {
        emptyBagText: "Your shopping bag is empty",
        totalText: "Total",
        promotionalText: "Promotional Savings",
        viewAndEditBag: "VIEW/EDIT BAG",
        sizeText: "Size:",
        qtyText: "Qty:",
        at: "at",
        negative: "-",
        itemText: "Item",
        itemsText: "Items",
        inBagText: "In Bag",
        salePrice: "Sale Price",
    },
    shoppingBag: {
        altText: "Shopping bag icon",
        youhaveText: "You have",
        itemText: "item",
        itemsText: "items",
        inYourBagText: "in your bag",
    },
    countryLangSelector: {
        headerTitle: "CHANGE COUNTRY",
        locationText: "Location",
        languageText: "Language",
        closeText: "Close",
        stayText: "Stay on Amido.co.uk",
        shopNowText: "Shop now",
        ROWText: "Can't find the country you're looking for?",
        ROWLinkText: "View All Countries",
    },
    countryChangeModal: {
        confirm: "CONTINUE",
        cancel: "CANCEL",
        bodyTextPt1: "Are you sure you want to navigate away from this site?",
        bodyTextPt2: "If you navigate away from this site you will leave your shopping bag and it’s contents.",
        titleText: "Confirm country change",
    },
    cookiePolicy: {
        title: "Cookie Policy",
        body: "We use cookies to provide you with the best posible experience. By continuing to use our site, you agree to our use of cookies.",
        link: "Find out more",
        linkText: "about managing your cookie settings.",
        closeIconAltText: "Close",
    },
    saleBagWarning: {
        title: "Careful",
        textOne: "You could lose your sale bag.",
        textTwo: "Would you like to return to our Sale Site?",
        primaryButtonText: "YES, RETURN TO THE SALE",
        secondaryButtonText: "NO, THANK YOU",
    },
    skipToMainContent: "Skip to main content",
    closeIconAltText: "Close",
}

export const mockState: State = {
    settings: {
        "userConsentManagement.enabled": false,
        variant: "default",
    },
    user: {firstName: "", loggedIn: false, userUpdated: false},
    autocomplete: mockAutoComplete,
    recents: mockRecents,
    shoppingBag: mockShoppingBag,
    search: mockSearch,
    template: "default",
    request: {
        headers: {"x-monorepo-territory": "gb"},
        url: "/",
        siteUrl: "fakeamido.com",
        timeMachineDate: "20201215",
        isInternationalCountry: false,
        fullTerritoryName: "United Kingdom",
        geolocationBaseUrl: "",
        geolocationVersion: 0,
        bloomReachCachingCookieList: "",
        bloomReachCachingEnabled: false,
        useTimeMachineCookie: false,
        currencyCode: "EUR",
        bloomreachGroupLocation: "southernhemisphere",
        bloomreachDomainKey: "GB",
        monetateSDK: false,
        accountMonetateSDK: "",
        cookieDomain: ".amido.com",
        showSaleWarningBag: false,
    },
    languages: mockLanguages,
    data: {...apiData, language: ""},
    textAlignment: "ltr",
    text: mockText,
    features: {
        SearchBar: {
            Value: "SimpleSearch",
            RecentSearch: {
                MaxItems: 12,
            },
            Autocomplete: {
                MaxItems: 10,
                ProductsMaxItems: {
                    xs: 8,
                    sm: 8,
                    md: 4,
                    lg: 4,
                    xl: 5,
                },
            },
        },
    },
    favourites: {enableFavourites: true, hasFavourites: true},
    countrySelector: {
        showCountrySelector: false,
        isActive: true,
        countriesList: null,
        selectedCountry: null,
        selectedLanguage: null,
        defaultCountryCode: null,
        showOverlay: true,
        showBFPOFlag: false,
        loaded: false,
        requestedCountryChange: false,
        countrySelectorEndpoint: "",
    },
}

export const mockTextAlignment = {Value: "ltr"}

export const mockTheme = {styles: mockRealmStyles, colours: mockColors, direction: "ltr"}
export const mockConfigureStore = configureMockStore([thunk])
const mockStore = mockConfigureStore(mockState)

export default mockStore
