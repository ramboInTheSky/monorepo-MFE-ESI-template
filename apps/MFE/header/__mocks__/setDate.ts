export const mockDateConstructor = (setDate: Date) => {
    Date = class extends Date {
        constructor() {
            super()
            return setDate
        }
    } as DateConstructor
}
