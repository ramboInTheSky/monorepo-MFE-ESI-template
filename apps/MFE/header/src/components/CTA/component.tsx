import React from "react"
import {Button as MUIButton} from "@mui/material"
import styled from "styled-components"
import {Theme} from "../../models/theme"

interface MuiButtonProps {
    themeType: string
    theme: Theme
    showOpacity: boolean
}

export const Button = styled(({theme, themeType, isCountrySelectorCTA, showOpacity, ...rest}) => (
    <MUIButton {...rest} />
))<MuiButtonProps>`
    && {
        width: 100%;
        height: 100%;
        font-family: ${({theme}) => `${theme.colours.font.primary.medium.family}, ${theme.colours.font.default}`};
        font-size: ${({isCountrySelectorCTA}) => (isCountrySelectorCTA ? "0.875rem" : "0.75rem")};
        font-weight: 500;
        letter-spacing: 1px;
        border: ${({theme, themeType}) => theme.colours.form[`button${themeType}`].border};
        color: ${({showOpacity, theme, themeType}) =>
            showOpacity ? theme.colours.form[`button${themeType}`].color : "black"};
        opacity: ${({showOpacity}) => (showOpacity ? 0.5 : 1)};
        border-radius: ${({theme, themeType}) => theme.colours.form[`button${themeType}`].radius};
        background-color: ${({theme, themeType}) => theme.colours.form[`button${themeType}`].background};
        :hover {
            background-color: ${({theme, themeType}) => theme.colours.form[`button${themeType}`].background};
        }
        :focus {
            background-color: ${({theme, themeType}) => theme.colours.form[`button${themeType}`].background};
        }
    }
    span {
        color: ${({theme, themeType}) => theme.colours.form[`button${themeType}`].color};
        text-transform: uppercase;
        letter-spacing: 0.0625rem !important;
    }
`
