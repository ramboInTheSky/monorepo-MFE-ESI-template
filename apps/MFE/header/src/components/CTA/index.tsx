import React from "react"
import {ThemeType} from "models/theme"

import {Button} from "./component"

export type CTAProps = {
    enable: boolean
    text: string
    url?: string
    themeType: ThemeType
    onClick?: () => void
    testId?: string
    dataGaV1?: string
    dataGaV2?: string
    dataGaV3?: string
    rel?: string
    isCountrySelectorCTA?: boolean
}

const CTA = ({
    enable,
    url,
    text,
    themeType,
    onClick,
    testId,
    dataGaV1,
    dataGaV2,
    dataGaV3,
    rel,
    isCountrySelectorCTA,
}: CTAProps) => {
    return (
        <Button
            href={url}
            onClick={onClick}
            disabled={!enable}
            showOpacity={!enable}
            themeType={themeType}
            disableRipple
            data-testid={testId}
            data-ga-v1={dataGaV1}
            data-ga-v2={dataGaV2}
            data-ga-v3={dataGaV3}
            rel={rel}
            isCountrySelectorCTA={isCountrySelectorCTA}
        >
            {text}
        </Button>
    )
}

export default CTA
