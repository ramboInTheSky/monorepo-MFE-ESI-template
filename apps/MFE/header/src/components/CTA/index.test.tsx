import React from "react"
import {render, screen} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {mockTheme} from "../../../__mocks__/mockStore"
import CTA, {CTAProps} from "."

describe("Components - CTA: ", () => {
    let props: CTAProps

    beforeEach(() => {
        props = {
            themeType: "Primary",
            enable: true,
            url: "/spiderman/",
            text: "sample",
        }
    })

    it("should match the snapshot", () => {
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <CTA {...props} />
            </SCThemeProvider>,
        )

        expect(asFragment()).toMatchSnapshot()
    })
    it("should have required attributes", () => {
        render(
            <SCThemeProvider theme={mockTheme}>
                <CTA {...props} />
            </SCThemeProvider>,
        )

        const button = screen.getByRole("link")
        expect(button).toHaveAttribute("href", props.url)
        expect(button).toHaveTextContent(props.text)
        expect(button).toHaveStyle("opacity:1")
    })
    it("should have required attributes when disabled", () => {
        render(
            <SCThemeProvider theme={mockTheme}>
                <CTA {...props} enable={false} />
            </SCThemeProvider>,
        )

        const button = screen.getByRole("link")
        expect(button).toHaveAttribute("href", props.url)
        expect(button).toHaveTextContent(props.text)
        expect(button).toHaveStyle("opacity: 0.5")
    })
})
