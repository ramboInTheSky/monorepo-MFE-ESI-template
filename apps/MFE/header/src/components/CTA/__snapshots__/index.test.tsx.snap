// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - CTA:  should match the snapshot 1`] = `
<DocumentFragment>
  .c0.c0 {
  width: 100%;
  height: 100%;
  font-family: "AzoSansMedium","Helvetica Neue",Arial,sans-serif;
  font-size: 0.75rem;
  font-weight: 500;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  border: 0;
  color: black;
  opacity: 1;
  border-radius: 0.25rem;
  background-color: #30a74b;
}

.c0.c0:hover {
  background-color: #30a74b;
}

.c0.c0:focus {
  background-color: #30a74b;
}

.c0 span {
  color: #ffffff;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.0625rem !important;
  -moz-letter-spacing: 0.0625rem !important;
  -ms-letter-spacing: 0.0625rem !important;
  letter-spacing: 0.0625rem !important;
}

<a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-text c0"
    href="/spiderman/"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      sample
    </span>
  </a>
</DocumentFragment>
`;
